### PR TITLE
original data point backend (update and delete)

### PR DESCRIPTION
**Pull request by rfontanarosa**
part of #1126

### CODE REVIEW

File: meta/assessment/originalDataPoint/originalDataPoint.ts [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
+// import { ODPEditStatus } from './odpEditStatus'
 import { ODPNationalClass } from './odpNationalClass'
-import { ODPValidation } from './odpValidation'
+// import { ODPValidation } from './odpValidation'
 
 export const ODPYears: Array<number> = Arrays.reverse(Arrays.range(1950, 2021))
 
 export interface OriginalDataPoint {
-  countryIso?: string
+  id: number
```
Comment (minotogna) [inline meta/assessment/originalDataPoint/originalDataPoint.ts:11]: this is optional

File: meta/assessment/originalDataPoint/originalDataPoint.ts [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
+  countryIso: string
```
Comment (minotogna) [inline meta/assessment/originalDataPoint/originalDataPoint.ts:12]: this is optional

File: server/controller/assessment/removeOriginalDataPoint.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import { Assessment, Cycle, OriginalDataPoint } from '@meta/assessment'
+
+export const removeOriginalDataPoint = async (
+  props: {
+    assessment: Assessment
+    assessmentCycle: Cycle
+    originalDataPoint: OriginalDataPoint
+  },
+  client: BaseProtocol = DB
+): Promise<OriginalDataPoint | null> => {
```
Comment (minotogna) [inline server/controller/assessment/removeOriginalDataPoint.ts:12]: why can this return null ? if we try to remove an odp that doesn't exist, we should get back an exception

File: server/controller/assessment/updateOriginalDataPoint.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import { Assessment, Cycle, OriginalDataPoint } from '@meta/assessment'
+
+export const updateOriginalDataPoint = async (
+  props: {
+    assessment: Assessment
+    assessmentCycle: Cycle
+    originalDataPoint: OriginalDataPoint
+  },
+  client: BaseProtocol = DB
+): Promise<OriginalDataPoint | null> => {
```
Comment (minotogna) [inline server/controller/assessment/updateOriginalDataPoint.ts:12]: why can this return null ? if we try to update an odp that doesn't exist, we should get back an exception

File: server/controller/assessment/updateOriginalDataPoint.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  const { assessment, assessmentCycle, originalDataPoint } = props
+
+  return client.tx(async (t) => {
+    const updatedOriginalDataPoint = await OriginalDataPointRepository.update(
+      { assessment, assessmentCycle, originalDataPoint },
+      t
+    )
+
```
Comment (minotogna) [inline server/controller/assessment/updateOriginalDataPoint.ts:20]: activity log is missing

File: server/controller/assessment/getOriginalDataPoint.ts [hunk 6]
```diff
@@ -3,14 +3,12 @@ import { AssessmentRepository, OriginalDataPointRepository } from '@server/repos
 import { OriginalDataPoint } from '@meta/assessment'
 
 export const getOriginalDataPoint = async (
-  props: { name: string; cycleName: string; odpId: string },
+  props: { name: string; cycleName: string; odpId: number },
   client: BaseProtocol = DB
-): Promise<OriginalDataPoint> => {
+): Promise<OriginalDataPoint | null> => {
```
Comment (minotogna) [inline server/controller/assessment/getOriginalDataPoint.ts:8]: why can this return null ?

File: server/repository/originalDataPoint/update.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      dataSourceMethods,
+      dataSourceReferences,
+      description,
+      nationalClasses,
+    },
+  } = params
+
+  const schemaName = Schemas.getNameCycle(assessment, assessmentCycle)
+
+  await client.oneOrNone<OriginalDataPoint | null>(
```
Comment (minotogna) [inline server/repository/originalDataPoint/update.ts:26]: why can this return null ? if we try to update an odp that doesn't exist, we should get back an exception

File: meta/assessment/originalDataPoint/originalDataPoint.ts (hunk 2 above)
Comment (rfontanarosa) [reply meta/assessment/originalDataPoint/originalDataPoint.ts:12]: Schermata 2022-02-14 alle 20 59 05

File: server/controller/assessment/removeOriginalDataPoint.ts (hunk 3 above)
Comment (rfontanarosa) [reply server/controller/assessment/removeOriginalDataPoint.ts:12]: ok

File: server/controller/assessment/updateOriginalDataPoint.ts (hunk 4 above)
Comment (rfontanarosa) [reply server/controller/assessment/updateOriginalDataPoint.ts:12]: ok

File: server/api/assessment/getOdp.ts [hunk 8]
```diff
@@ -8,7 +8,8 @@ export const AssessmentGetOdp = {
     express.get(ApiEndPoint.Assessment.OriginalDataPoint.one(), async (req: Request, res: Response) => {
       try {
         const { name, cycleName, odpId } = req.params
-        const odp = await AssessmentController.getOriginalDataPoint({ name, cycleName, odpId })
+        const id = +odpId
```
[the remark this answers is not in the document]
Comment (rfontanarosa) [reply server/api/assessment/getOdp.ts:11]: odpId: Number(odp)

File: server/repository/originalDataPoint/create.ts [hunk 9]
```diff
@@ -10,12 +10,41 @@ export const create = async (
...
+    assessmentCycle,
+    originalDataPoint: {
+      countryIso,
+      year,
+      dataSourceAdditionalComments,
+      dataSourceMethods,
+      dataSourceReferences,
+      description,
+      nationalClasses,
+    },
```
[the remark this answers is not in the document]
Comment (rfontanarosa) [reply server/repository/originalDataPoint/create.ts:24]: that is not a definition but a destructuration

File: server/controller/assessment/updateOriginalDataPoint.ts (hunk 5 above)
Comment (rfontanarosa) [reply server/controller/assessment/updateOriginalDataPoint.ts:23]: done in update and remove

**Review by minotogna — APPROVED** [review]

**Pull request merged**